integration-tests/tests/run-workflow.js

```diff
@@ -57,33 +59,52 @@ const launchWorkflowAndWaitForSuccess = async page => {
   ])
   // stopLoggingPageAjaxResponses()
 
+  // If this table doesn't exists, something is wrong. Fail test now.
+  await page.waitForXPath('//*[@role="table" and @aria-label="submission details"]', { visible: true })
+
+  // Workflow status list:
+  // Queued -> Submitted -> Launching -> Running -> Succeeded or Failed
   const start = Date.now()
+
+  // Long enough for the submission details to be Running or Failed.
+  // Workflows might sit in "Submitted" for a long time if there is a large backlog of workflows in the environment
+  await Promise.race([
+    findInGrid(page, 'Failed', { timeout: 5 * 60 * 1000 }),
+    findInGrid(page, 'Running', { timeout: 5 * 60 * 1000 })
+  ])
+
+  // If status is Failed, fails test now
+  try {
+    await findInGrid(page, 'Failed', { timeout: 1000 })
```

Comment:
I think this can be combined with the previous Promise.race with something like:

```js
const workflowStatus = await Promise.race([
  findInGrid(page, 'Failed', { timeout: 5 * 60 * 1000 }).then(() => 'Failed'),
  findInGrid(page, 'Running', { timeout: 5 * 60 * 1000 }).then(() => 'Running')
])
```

Reply:
like this?
<img width="507" alt="Screen Shot 2022-05-20 at 1 47 26 PM" src="https://user-images.githubusercontent.com/35533885/169583916-75dacbc0-796a-482d-912c-97fb82c1160d.png">

Reply:
Yes, I think that should work.